client-runtime/src/main/java/com/microsoft/rest/v2/RestProxy.java

```diff
@@ -109,6 +112,28 @@ else if (returnTypeToken.isSubtypeOf(Single.class)) {
         }
         else {
             final HttpResponse response = httpClient.sendRequest(request);
+
+            final int responseStatusCode = response.statusCode();
+            if (!methodParser.isExpectedResponseStatusCode(responseStatusCode)) {
+                final Class<? extends RestException> exceptionType = methodParser.exceptionType();
+                try {
+                    final Class<?> exceptionBodyType = methodParser.exceptionBodyType();
+                    final Constructor<? extends RestException> exceptionConstructor = exceptionType.getConstructor(String.class, HttpResponse.class, exceptionBodyType);
+
+                    String responseContent = null;
+                    try {
+                        responseContent = response.bodyAsString();
+                    } catch (IOException ignored) {
+                    }
+
+                    final Object exceptionBody = responseContent == null ? null : serializer.deserialize(responseContent, exceptionBodyType);
```

Comment:
If there is an exception thrown by de-serializer while parsing the string `responseContent`, can we ensure message part of `IOException` contains `responseContent`?

Reply:
i mean the message part of `IOException` we throw below (line#133)

Reply:
I've added responseContent to the IOException's message if the responseContent isn't null or empty.